docs/conversion_examples_gallery/behavior/deeplabcut.rst

```diff
@@ -0,0 +1,25 @@
+DeepLabCut data conversion
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Convert DeepLabCut imaging data to NWB using :py:class:`~neuroconv.datainterfaces.behavior.deeplabcut.deeplabcutdatainterface.DeepLabCutInterface`.
+
+.. code-block:: python
+
+    >>> from datetime import datetime
+    >>> from dateutil import tz
+    >>> from pathlib import Path
+    >>> from neuroconv import DeepLabCutInterface
+    >>>
+    >>> file_path = BEHAVIOR_DATA_PATH / "DLC" / "m3v1mp4DLC_resnet50_openfieldAug20shuffle1_30000.h5"
+    >>> config_file_path = BEHAVIOR_DATA_PATH / "DLC" / "config.yaml"
+    >>>
+    >>> interface = DeepLabCutInterface(file_path=file_path, config_file_path=config_file_path, subject_name="ind1", verbose=False)
+    >>> metadata = interface.get_metadata()
+    >>> # For data provenance we add the time zone information to the conversion
+    >>> session_start_time = datetime(2020, 1, 1, 12, 30, 0, tzinfo=tz.gettz("US/Pacific"))
+    >>> metadata.get("NWBFile", dict()).update(session_start_time=session_start_time)
+    >>> # Choose a path for saving the nwb file and run the conversion
+    >>> interface.run_conversion(nwbfile_path=path_to_save_nwbfile, metadata=metadata)
+    >>> # If the conversion was successful this should evaluate to ``True`` as the file was created.
+    >>> Path(path_to_save_nwbfile).is_file()
```

Comment:
Where is `path_to_save_nwbfile` defined? The purpose of these galleries is to provide easily copied code to be sent to people that want to paste it, point to their data, and run.

I get that we have to use the global `BEHAVIOR_DATA_PATH` to achieve that testing feature with the pytest fixtures, but other variable references should be standalone and/or more easily modifiable - even by first-time users of Python that otherwise have general understanding of coding

Reply:
I guess relates to https://github.com/catalystneuro/neuroconv/pull/24#discussion_r933229021

so the user is intended to just know to change the `path_to_save_nwbfile` variable to some place on their system? Feel like that's not super apparent or as user-friendly as it could be

Reply:
It was like this before
```python
    >>> nwbfile_path = f"{path_to_save_nwbfile}"
    >>> interface.run_conversion(nwbfile_path=nwbfile_path, metadata=metadata)
```

The idea was to indicate that it should be a string with the path to the nwbfile but then I changed it above after Ben's suggestion and I am less sure it was that clear either. Do you have any suggestion on how to make this more user friendly?

Reply:
The simplest thing we would want to share via copy/paste is

```python
from neuroconv import SpikeGLXRecordingInterface

spikeglx_file_path = ECEPHYS_DATA_PATH  / "spikeglx_file.ap.bin"
nwbfile_path = "path/to/save/my_nwbfile.nwb"

interface = SpikeGLXRecordingInterface(file_path=spikeglx_file_path)
interface.run_conversion(nwbfile_path=nwbfile_path)
```

from https://github.com/catalystneuro/neuroconv/blob/main/src/neuroconv/nwbconverter.py#L163-L165 they shouldn't even have to get the metadata for interfaces that support automatic setting of session start time. For those that do something like

```python
from datetime import datetime
from dateutil import tz
from neuroconv import NeuralynxRecordingInterface

session_start_time = datetime(2020, 1, 1, 12, 30, 0, tzinfo=tz.gettz("US/Pacific"))  # Y, M, D, H, M, S
neuralynx_folder_path = "path/to/neuralynx_folder"
nwbfile_path = "path/to/save/my_nwbfile.nwb"

interface = NeuralynxRecordingInterface(folder_path=neuralynx_folder_path )
metadata = interface.get_metadata()
metadata["NWBFile"].update(session_start_time=session_start_time)
interface.run_conversion(nwbfile_path=nwbfile_path, metadata=metadata)
```

but of course more than anything, we should be consistent across the gallery. It looks like most of the currently galleries use the `f"{path_to}"` convention, which I agree with Ben looks a little awkward especially if the user as no idea what an f-string is. I do think that the `ECEPHYS_DATA_PATH` global is fine to use as it is at the top and all in capitals, it should be more apparent to the user to change that for their data. Though we could also leave a comment noting that if you disagree.

I'd also combine all definitions of paths or datetimes needed prior to calling the 2-4 line code sequence to run things so it's more apparent what the 'input' vs. 'output' is. So the order is (i) do imports, (ii) define paths as strings or mixtures of the global data paths needed for fixtures, (iii) 2-4 lines to run the standalone conversion through the data interface.

Reply:
>  metadata.get("NWBFile", dict()).update(session_start_time=session_start_time)

As an aside (just noticing) I would also say that every `get_metadata` call should return an `NWBFile` level to the dictionary structure, if it doesn't then we should add that in where appropriate. The recording interfaces do this automatically: https://github.com/catalystneuro/neuroconv/blob/5f55cdb49d3b613d06fc8cfcde1b2faac0081e41/src/neuroconv/tools/spikeinterface/spikeinterface.py#L82-L85

but the DLC is standalone so maybe it doesn't have this yet?

Reply:
If you find a way to make a notation like "/this/is/a/path" that indicates clearly that the user should change that by a path and work with the testing that would be great. 

Concerning using scripting / scientific programming style for the tests (putting all the definitions at the beginning) I personally don't like it and I find it more difficult to follow but I honestly don't feel strong about it. Let's discuss this issues in the following team meeting as I feel this just about getting to a consensus and sticking with it.

About the metadata comment that can be added here. While doing the conversion gallery I found some that did not have this structure. I don't remember which ones. I agree that we should correct that.

Reply:
> 

We can do it now for the main item in discussion of the `nwbfile_path` ass illustrated above. We don't need the NWBFile to be saved anywhere specifically and for any user it would just be the working directory (or absolute path if they used that on their system).

The data paths are a bit more stringent, I do understand. I guess just have to keep those as-is for now. Down the road we could do something fancy like form a symlink to a made-up folder `/path/to/data` or something but that could also be overkill. My main gripe is just with setting the `nwbfile_path` is the simplest, clearest way which IMO is just to set it equal to a simple string (not fstring or pytest fixture variable)

Reply:
Still, no reason to hold up the DataInterface PR for this. Can discuss in team meeting to achieve consistency

Reply:
Yes I agree this is a good convo and also that this should not hold up the DLC support. 

I like the idea of `get_metadata` always including an NWBFile. We should adjust the json-schema to make the NWBFile required.

I see that we are using a fixture that creates a temp directory, but I I agree that few of our users are likely to be comfortable with fixtures and it might confuse them. Would it be a problem to just save these files in the local directory with just something like:

```python
    >>> interface.run_conversion(nwbfile_path='output_nwbfile.nwb', metadata=metadata)
```
?

Reply:
> Would it be a problem to just save these files in the local directory with just something like:
> interface.run_conversion(nwbfile_path='output_nwbfile.nwb', metadata=metadata)

That's what I'm proposing, though @h-mayorquin likes to decouple the assignment into two steps

```python
nwbfile_path = "output_nwbfile.nwb"
interface.run_conversion(nwbfile_path=nwbfile_path, metadata=metadata)
```

to avoid having a single code line that does two operations (both declaring save path and running conversion).

Personally though I also like having as few a lines as code as possible for these kind of things so I'd also like nothing more than to have 7 lines (above) reduced to 4

```python
interface = NeuralynxRecordingInterface(folder_path="path/to/neuralynx_folder")
metadata = interface.get_metadata()
metadata["NWBFile"].update(session_start_time=datetime(2020, 1, 1, 12, 30, 0, tzinfo=tz.gettz("US/Pacific")))
interface.run_conversion(nwbfile_path="path/to/save/my_nwbfile.nwb", metadata=metadata)

```

Reply:
>That's what I'm proposing, though @h-mayorquin likes to decouple the assignment into two steps
I am fine with Ben proposal of having as a one liner in this case. It is true that I like explicit assignments in code for both auto-documentation and debugging purposes but here just the variable name is clear enough so the correction that Ben did above is already committed. The declaration in this case though was for a different purpose as the variable is needed once again for testing that the conversion run correctly.

I also think that Cody proposal of using a save path that goes something like "this/is/where/stuff/should/be/saved" is OK as well. Although I honestly don't see a lot of difference between that and have a comment there indicating that the user should modify the path with their desired save destination. If you both want to change it I would still think that we should do it for all them at the same time. I was not explicit above about the constrains of testing this way though:
1) Each test file path should be unique or go to a different folder so when you run all the tests together in the CI or in your local machine you don't get input output errors.
2) You might want to create a script that cleans all the files at the end for when you run the testing suit on your local machine (not sure it creates problems in CI, I don't think so).

The fixtures are not intended for the users to modify or even be aware of, the goal of the fixtures is so automatic testing assures that the documentation keeps working. My goal was always that the users should modify all the path themselves. As an example, this is the one in neuralynx:
```python
    >>>  # Choose a path for saving the nwb file and run the conversion
    >>> nwbfile_path = f"{path_to_save_nwbfile}"  # Modify this to suit your needs to something like: "./saved_file.nwb"
    >>> interface.run_conversion(nwbfile_path=nwbfile_path, metadata=metadata)
    >>>
    >>> # If the conversion was successful this should evaluate to ``True`` as the file was created.
    >>> Path(nwbfile_path).is_file()
    True
 ```

Reply:
So, to summarize, please feel free to add if I am missing something those are the issues:
1) Make the saving paths unique so they don't need to be modified.
2) Decide between explicit declaration vs implicit (I favor explicit but I think you two favor saving lines so I am fine with reducing it here for all the other interfaces.
3) Script format where everything is declared at the beginning is another option if we chose to keep explicit declaration.